.gitlab/e2e/e2e.yml

```diff
@@ -178,33 +208,37 @@ new-e2e-agent-subcommands:
       - EXTRA_PARAMS: --run TestLinuxCheckSuite
 
 new-e2e-language-detection:
-  extends: .new_e2e_template
+  extends: .new_e2e_template_needs_deb_x64
   rules: !reference [.on_language-detection_or_e2e_changes_or_manual]
-  needs: ["deploy_deb_testing-a7_x64", "deploy_windows_testing-a7"]
   variables:
     TARGETS: ./tests/language-detection
     TEAM: processes
 
 new-e2e-npm:
   extends: .new_e2e_template
   rules: !reference [.on_npm_or_e2e_changes_or_manual]
-  needs: ["deploy_deb_testing-a7_x64", "deploy_windows_testing-a7", "qa_agent"]
+  needs:
+    - qa_agent
+    - deploy_deb_testing-a7_x64
+    - deploy_rpm_testing-a7_x64
+    - deploy_windows_testing-a7
   variables:
     TARGETS: ./tests/npm
     TEAM: network-performance-monitoring
 
 new-e2e-aml:
-  extends: .new_e2e_template
+  extends: .new_e2e_template_needs_deb_windows_x64
   rules: !reference [.on_aml_or_e2e_changes_or_manual]
-  needs: ["deploy_deb_testing-a7_x64", "deploy_windows_testing-a7"]
   variables:
     TARGETS: ./tests/agent-metric-logs
     TEAM: agent-metric-logs
 
 new-e2e-cws:
   extends: .new_e2e_template
   rules: !reference [.on_cws_or_e2e_changes_or_manual]
-  needs: ["deploy_deb_testing-a7_x64", "deploy_windows_testing-a7", "qa_agent", "qa_cws_instrumentation"]
+  needs:
```

Comment:
I think we need to keep the dependency on `qa_agent` because we need the agent image here in CWS test: https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/tests/cws/fargate_test.go#L100

Reply:
can confirm we need the agent yes,  but removing the windows needs is good !

Reply:
thanks